CONTRIBUTING.md

```diff
@@ -0,0 +1,662 @@
+      };
+    in
+    "${configFile}";
+}
+```
+
+Setting options through an existing NixOS or Home Manager module is preferable
+to generating whole files, since users will have the option of overriding things
+individually.
+
+Also note that reading generated files with `builtins.readFile` can be very slow
+and should be avoided.
+
+### How to apply other things
+
+For everything else, like fonts and wallpapers, you can just take option values
+directly from `config`. See the reference pages for a list of options.
+
+### Metadata
+
+Metadata is stored in `/modules/«module»/meta.nix`. The following attributes are
+available under `meta`:
+
+- `name`: required human-readable string name.
+
+- `homepage`: homepage string or attribute set of homepage strings, depending
+  on the number of homepages:
+
+  - ```nix
+    homepage = "https://github.com/nix-community/stylix";
+    ```
+
+  - ```nix
+    homepage = {
+      Nix = "https://github.com/NixOS/nix";
+      Nixpkgs = "https://github.com/NixOS/nixpkgs";
+    };
+    ```
+
+  The attribute names are used as hyperlink text and the attribute values are
+  used as URLs.
+
+- `maintainers`: required list of maintainers. See [Maintainers](#maintainers)
+  section.
+
+- `description`: optional markdown string for extra documentation.
+
+#### Maintainers
+
+New modules must have at least one maintainer.
+
+If you are not already listed in the Nixpkgs `/maintainers/maintainer-list.nix`
+maintainer list, add yourself to `/stylix/maintainers.nix`.
+
+Add yourself as a maintainer in one of the following ways, depending on the
+number of maintainers:
+
+- ```nix
+  { lib, ... }:
+  {
+    maintainers = [ lib.maintainers.danth ];
+  }
+  ```
+
+- ```nix
+  { lib, ... }:
+  {
+    maintainers = with lib.maintainers; [
+      danth
+      naho
+    ];
+  }
+  ```
+
+The main responsibility of module maintainers is to update and fix their
+modules.
+
+### Documentation
+
+Documentation for options is automatically generated. To improve the quality of
+this documentation, ensure that any custom options created using `mkOption` are
+given an appropriate `type` and a detailed `description`. This may use Markdown
+syntax for formatting and links.
+
+For modules needing more general documentation, add a `description` to
+`modules/«module»/meta.nix`:
+
+```nix
+description = ''
+  Consider describing which applications are themed by this module (if it's not
+  obvious from the module name), how the applications need to be installed for
+  the module to work correctly, which theming items are supported (colors,
+  fonts, wallpaper, ...), and any caveats the user should be aware of.
+'';
+```
+
+This will be inserted before the automatically generated list of options.
+
+You can build and view the documentation by running `nix run .#doc`, which will
+build the documentation, start a localhost web-server, and open it in your
+browser.
+
+See also: [Development environment → Documentation](#documentation-1)
```

Comment:
Personally, I find the `→` character confusing:

```suggestion
See also: [Development environment > Documentation](#documentation)
```

More flowy phrasings would be:

- ```suggestion
  See also the [documentation on the development environment](#documentation)
  ```

- ```suggestion
  See also the ["Development environment" documentation](#documentation)
  ```

I prefer the "documentation on the development environment" variant.